### PR TITLE
feat: Developer Guide - GUI - DBeaver

### PR DESCRIPTION
**Pull request by Icemap**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by Oreoxmt** [comment]
@Icemap https://github.com/Icemap/docs/pull/7 PTAL

File: develop/dev-guide-gui-dbeaver.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+> **Note:**
+>
+> This tutorial is compatible with TiDB Serverless, TiDB Dedicated, and TiDB Self-Hosted.
+
+## Prerequisites
+
+To complete this tutorial, you need:
+
+- [DBeaver Community **23.0.3** or higher](https://dbeaver.io/download/).
```
Comment (Oreoxmt) [inline develop/dev-guide-gui-dbeaver.md:20]: The latest tested version listed in `dev-guide-third-party-support.md` is 23.0.1. Should we update this to match the latest tested version?
https://github.com/pingcap/docs/blob/88124e9776b11e2c1799b2d9a5cfa08c0a7ea517/develop/dev-guide-third-party-support.md?plain=1#L182

File: develop/dev-guide-gui-dbeaver.md
Comment (Oreoxmt) [inline develop/dev-guide-gui-dbeaver.md:1]: Preview: https://pingcap-docs-website-scaffold-git-preview-pingca-518ba9-oreoxmt.vercel.app/tidb/dev/dev-guide-gui-dbeaver

File: develop/dev-guide-gui-dbeaver.md (hunk 1 above)
Comment (Icemap) [reply develop/dev-guide-gui-dbeaver.md:20]: good catch!

**Review by Oreoxmt — APPROVED** [review]

**Review by qiancai — APPROVED** [review]
LGTM

**Pull request merged**